cardinal/world_persona.go

```diff
@@ -0,0 +1,43 @@
+package cardinal
+
+import (
+	"errors"
+	"pkg.world.dev/world-engine/cardinal/filter"
+	"pkg.world.dev/world-engine/cardinal/persona/component"
+	"pkg.world.dev/world-engine/cardinal/types"
+)
+
+var (
+	ErrPersonaTagHasNoSigner        = errors.New("persona tag does not have a signer")
+	ErrCreatePersonaTxsNotProcessed = errors.New("create persona txs have not been processed for the given tick")
+)
+
+// GetSignerForPersonaTag returns the signer address that has been registered for the given persona tag after the
+// given tick. If the engine's tick is less than or equal to the given tick, ErrorCreatePersonaTXsNotProcessed is
+// returned. If the given personaTag has no signer address, ErrPersonaTagHasNoSigner is returned.
+func (w *World) GetSignerForPersonaTag(personaTag string, tick uint64) (addr string, err error) {
```

Comment:
i feel like this logic should be moved to the persona plugin.

Reply:
this is ideally what i wanted to do, but the server code depends on this via WorldContext, so moving it to plugin would cause circular dependency.

Reply:
we should move plugins to their own go.mod, as plugin authors would do to write their own

ill make a ticket for it

Reply:
wouldn't that also cause circular dependencies? 

`cardinal` depends on `arbitrary_plugin`
`arbitrary_plugin` depends on `cardinal`

Reply:
nope - circular dependency happens for packages, not go modules